### PR TITLE
Baris/update docs with class api

### PR DESCRIPTION
**Pull request by bcdurak**
## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [x] If my change requires a change to the documentation, I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Describe changes
New pages for the documentation

### CODE REVIEW

File: docs/book/guides/class-based-api/chapter-1.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
+---
+description: Create your first step.
+---
+
+If you want to see the code for this chapter of the guide, head over to the [GitHub](https://github.com/zenml-io/zenml/tree/main/examples/low_level_guide/chapter_1.py).
```
Comment (alexej-zenml) [inline docs/book/guides/class-based-api/chapter-1.md:5]: is this link correct? It seems like this is the functional code with decorators at that link

File: docs/book/guides/class-based-api/chapter-1.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        return pd.read_csv(
+            filepath_or_buffer=config.path,
+            sep=config.sep,
+            header=config.header,
+            names=config.names,
+            index_col=config.index_col,
+        )
+```
+
+Importing things to note:
```
Comment (alex-zenml) [inline docs/book/guides/class-based-api/chapter-1.md:47]:
```suggestion
Importing: things to note:
```

**Review by htahir1 — CHANGES_REQUESTED** [review]
Looks really good but should we not have a corresponding example like here: https://github.com/zenml-io/zenml/tree/main/examples/low_level_guide?

Also I think we should merge @schustmi branch first and then this one because the name changes might conflict :-)

File: docs/book/guides/class-based-api/chapter-2.md [hunk 3]
```diff
@@ -0,0 +1,257 @@
+---
+description: Split and preprocess your dataset
+---
+
+If you want to see the code for this chapter of the guide, head over to the [GitHub](https://github.com/zenml-io/zenml/tree/main/examples/low_level_guide/chapter_2.py).
```
Comment (alexej-zenml) [inline docs/book/guides/class-based-api/chapter-2.md:5]: link will need to be updated

File: docs/book/guides/class-based-api/chapter-3.md [hunk 4]
```diff
@@ -0,0 +1,231 @@
+---
+description: Train some models.
+---
+
+If you want to see the code for this chapter of the guide, head over to the [GitHub](https://github.com/zenml-io/zenml/tree/main/examples/low\_level\_guide/chapter\_3.py).
+
+Finally, we can train and evaluate our model. For this we decide to utilize the two base class, the `BaseTrainerStep`
```
Comment (alex-zenml) [inline docs/book/guides/class-based-api/chapter-3.md:7]:
```suggestion
Finally, we can train and evaluate our model. For this we decide to utilize the two base classes, the `BaseTrainerStep` 
```

File: docs/book/guides/class-based-api/chapter-3.md [hunk 5]
```diff
@@ -0,0 +1,231 @@
+---
+description: Train some models.
+---
+
+If you want to see the code for this chapter of the guide, head over to the [GitHub](https://github.com/zenml-io/zenml/tree/main/examples/low\_level\_guide/chapter\_3.py).
```
Comment (alexej-zenml) [inline docs/book/guides/class-based-api/chapter-3.md:5]: link will need to be updated

**Review by alexej-zenml — APPROVED** [review]

File: docs/book/guides/class-based-api/chapter-3.md [hunk 6]
```diff
@@ -0,0 +1,231 @@
...
+
+        predictions = model.predict(dataset)
+        predicted_classes = [1 if v > 0.5 else 0 for v in predictions]
+
+        report = classification_report(labels, predicted_classes, output_dict=True)
+
+        return report
+```
+
+Important thing to note:
```
Comment (alex-zenml) [inline docs/book/guides/class-based-api/chapter-3.md:112]:
```suggestion
Important things to note:
```

File: docs/book/guides/class-based-api/chapter-3.md [hunk 7]
```diff
@@ -0,0 +1,231 @@
...
+
+You get the following output:
+
+```bash
+Pipeline `mnist_pipeline` has 1 run(s)
+The first run has 4 steps.
+The `tf_evaluator step` returned an accuracy: 0.9100000262260437
+```
+
+Wow, we just trained our first model! But have not stopped yet. What if did not want to use TensorFlow? Let's swap out our trainers and evaluators for different libraries.
```
Comment (alex-zenml) [inline docs/book/guides/class-based-api/chapter-3.md:231]:
```suggestion
Wow, we just trained our first model! But we're not done yet. Suppose you don't want to use TensorFlow... Let's swap out our trainers and evaluators for different libraries.
```

File: docs/book/guides/functional-api/README.md [hunk 8]
```diff
@@ -4,7 +4,7 @@ description: Build production ML pipelines from the simple step interface.
 
 # Functional API
 
-The Functional ZenML API is defined by the primitive `@step` and `@pipeline` decorators. These make it easy to quickly take your functional code into a zenml pipeline. If you want more control over the different steps the [Class based API](../high-level-api/)  is the way to go.&#x20;
+The Functional ZenML API is defined by the primitive `@step` and `@pipeline` decorators. These make it easy to quickly take your functional code into a zenml pipeline. If you want more control over the different steps the [Class based API](../class-based-api/)  is the way to go.&#x20;
```
Comment (alex-zenml) [inline docs/book/guides/functional-api/README.md:7]:
```suggestion
The Functional ZenML API is defined by the primitive `@step` and `@pipeline` decorators. These make it easy to quickly take your functional code into a ZenML pipeline. If you want more control over the different steps the [Class based API](../class-based-api/)  is the way to go.&#x20;
```

File: docs/book/toc.md [hunk 9]
```diff
@@ -8,15 +8,20 @@
...
-  * [Train & Evaluate](guides/low-level-api/chapter-3.md)
-  * [Caching](guides/low-level-api/chapter-4.md)
-  * [Materialize artifacts](guides/low-level-api/chapter-5.md)
-  * [Import dynamic data](guides/low-level-api/chapter-6.md)
-  * [Deploy to production](guides/low-level-api/chapter-7.md)
-* [Class based API](guides/high-level-api/README.md)
+* [Functional API](guides/functional-api/README.md)
+  * [Create a step](guides/functional-api/chapter-1.md)
+  * [Normalize data](guides/functional-api/chapter-2.md)
+  * [Train & Evaluate](guides/functional-api/chapter-3.md)
```
Comment (alex-zenml) [inline docs/book/toc.md:14]:
```suggestion
  * [Train & evaluate](guides/functional-api/chapter-3.md)
```

File: docs/book/toc.md [hunk 10]
```diff
@@ -8,15 +8,20 @@
...
+  * [Caching](guides/functional-api/chapter-4.md)
+  * [Materialize artifacts](guides/functional-api/chapter-5.md)
+  * [Import dynamic data](guides/functional-api/chapter-6.md)
+  * [Deploy to production](guides/functional-api/chapter-7.md)
+* [Class-based API](guides/class-based-api/README.md)
+  * [Create a step](guides/class-based-api/chapter-1.md)
+  * [Normalize data](guides/class-based-api/chapter-2.md)
+  * [Train & Evaluate](guides/class-based-api/chapter-3.md)
```
Comment (alex-zenml) [inline docs/book/toc.md:22]:
```suggestion
  * [Train & evaluate](guides/class-based-api/chapter-3.md)
```

**Review by alex-zenml — CHANGES_REQUESTED** [review]
Really loved this one. And I learnt a thing or two along the way! Some nits below...

File: docs/book/guides/class-based-api/chapter-1.md (hunk 1 above)
Comment (bcdurak) [reply docs/book/guides/class-based-api/chapter-1.md]: updated the link

File: docs/book/guides/class-based-api/chapter-2.md (hunk 3 above)
Comment (bcdurak) [reply docs/book/guides/class-based-api/chapter-2.md]: updated the link

File: docs/book/guides/class-based-api/chapter-3.md (hunk 5 above)
Comment (bcdurak) [reply docs/book/guides/class-based-api/chapter-3.md]: updated the link

File: docs/book/guides/class-based-api/chapter-3.md (hunk 7 above)
Comment (bcdurak) [reply docs/book/guides/class-based-api/chapter-3.md]: I have taken out this part as the next stages of the docs is still in progress.

File: docs/book/guides/functional-api/README.md (hunk 8 above)
Comment (bcdurak) [reply docs/book/guides/functional-api/README.md]: Thanks, I have pushed the fixed version.

File: docs/book/toc.md (hunk 9 above)
Comment (bcdurak) [reply docs/book/toc.md]: Thanks, I have pushed the fixed version.

File: docs/book/toc.md (hunk 10 above)
Comment (bcdurak) [reply docs/book/toc.md]: Thanks, I have pushed the fixed version.

**Review by AlexejPenner — APPROVED** [review]

**Pull request merged**